vm/actor/tests/puppet_actor_test.rs

```diff
@@ -86,4 +86,17 @@ fn serialize_test() {
 
     // Should fail becuase vec is no longer empty
     assert!(Serialized::serialize(v).is_err());
+
+    let mut v: Vec<Option<FailToMarshalCBOR>> = vec![];
+
+    assert!(Serialized::serialize(&v).is_ok());
```

Comment:
yeah this proves that for the transaction marshal failure call will not fail during serialization